app/Controllers/CsvImportController.php

```
@@ -1506,35 +1737,50 @@ classificazione_dewey, stato, created_at
+        // count); raise it to a still-DoS-safe ceiling and log when it bites so a
+        // truncation is diagnosable rather than invisible.
         if ($copie < 1) {
             $copie = 1;
-        } elseif ($copie > 100) {
-            $copie = 100;  // Max 100 copie per libro da CSV import
+        } elseif ($copie > 2000) {
+            \App\Support\SecureLogger::warning('CsvImportController: copie_totali troncato all\'import', [
+                'titolo' => $titolo,
+                'requested' => $copie,
+                'capped_to' => 2000,
+            ]);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\SecureLogger' in method 'insertBook'. (undefined)

(StaticAccess)